### DISCUSSION TITLE
2690v2 disabling cpu affinity makes 5-10% imrovement

### DISCUSSION THREAD

**Discussion by mocksoul** [Chia-Network/bladebit#10] [Ideas]
Hi!

I tried to disable cpu affinity completely and that makes up to 10% improvement (usually around 5-7%).
linux `5.10.27` custom built kernel

`2x2690v2 3.3GHz OC, 512G RAM`, numa memory interleaving enabled (saw speedups in 2 phase =)). 20 cores total (40 threads). ~25-50% cpu other processes usage.

Probably would be good idea to add `--no-cpu-affinity` switch at least? What do you think?

update: ctx switches dropped 50% by this & numa interleave change in this system

![image](https://user-images.githubusercontent.com/307574/126037731-1ac50f27-fa8c-4ab1-9e37-c97d881a0a9b.png)

update: made quick test on 80 core system (cascade lake, cloud). Turning off cpu affinity leads to huge system kernel use in phase1 :)). It is 3-4 times slower with cpu affinity off. So, not all systems will benefit on turning off affinity..

**Comment by harold-b** [comment]
Agreed on the disabling cpu affinity switch, and I was going to add it to that commit but decided against it as it already had too many things. 

I’ll see about doing another update tonight.

really appreciate these reports!

Unrelated: Were you able resolve the pool errors with the latest update?

**Reply by mocksoul** [reply]
> Unrelated: Were you able resolve the pool errors with the latest update?

yes. Big thumbs up for that!

**Comment by koiesmith** [comment]
Mocksoul - by chance can you share your commands you used to turn settings on and off in Linux and run the benchmarks. I have lots of servers with v2,v3, and v4 procs and I'm trying to optimize them but learning Linux at the same time!!

**Reply by harold-b** [reply]
FYI the current `develop` has support for turning off cpu affinity via the `--no-cpu-affinity` switch. It also include a substantial optimization for sorting on Phase 1 (typically getting 20+ seconds saved, depending on the system) so you might want to test with that branch.
